### ISSUE TITLE
Is there a way to set a default environment or profile for the default bash shell?

### ISSUE THREAD

**Issue by shakefu** [actions/runner#1530]
From what I can tell, the runner will use a default set of bash arguments including `--norcfile` and `--noprofile`, which means it won't load any Shell scripts like `nvm` or `tfenv`. 

I see it's possible to override the shell invocation per-workflow, but what I really would like is some way to set the default shell invocation globally from within a runner so that we can get a fully initialized login/rc shell, in order to be able to use these tools easily, e.g. `run: nvm exec 14.17.3 ci.js`.

Specifically to our use case we have several of these _*env_ tools in the `/etc/profile` which works fine as long as bash is invoked normally, but of course without the login/profile processing, the shims PATHs aren't set and the function based commands (nvm) don't exist.

**Comment by igorcosta** [comment]
I tried that, and I managed to get working by setting job outputs to the env >> GITHUB_ENV "workaround" and set a time between jobs within 5min to execute the next job. It works for me.

**Comment by shakefu** [comment]
Yeah ideally I'd really like a way to configure the defaults for the shell in the runner invocation itself. There is a set of default arguments for the shell:
https://github.com/actions/runner/blob/be9632302ceef50bfb36ea998cea9c94c75e5d4d/src/Runner.Worker/Handlers/ScriptHandlerHelpers.cs
... but sadly it's not configurable anywhere. I suppose if we really had to we could fork this repository and make changes if the maintainer team don't see a broader use case here...

**Comment by nikola-jokic** [comment]
Hey everyone,

I will mark this as a feature request and add this to the board.

**Comment by dpc** [comment]
OK, so here is how I make it work:

Start the runner via script (I use `crontab -e` and `@reboot` to start it on boot):

```bash
#!/usr/bin/env bash

cd $HOME/actions-runner
env PATH="$HOME/actions-runner-hack/:$PATH" ./run.sh
```

As you can see it adds an extra dir to the PATH. In this dir, we put a `bash` wrapper script. Since actions runner is using just `bash` it will pick the first one available on the `PATH`.

And that hacked `bash` looks like this:

```bash
#!/bin/bash

if [ -z "$SOURCED_NIX_ALREADY" ]; then
  export SOURCED_NIX_ALREADY=yes
  . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
fi

exec /bin/bash "$@"
```

So basically - it can do any initialization that we need (in this case sourcing nix) and call the real `/bin/bash`. If need we could do stuff like skipping `--noprofile` etc. , but it seems unnecessary for my use-case.

Now everything just works :crossed_fingers:, without any modifications to github actions scripts, with all hacking done on the runner machine.

**Comment by theonetheycallneo** [comment]
@dpc any way to get this to work with svc?

**Comment by dpc** [comment]
> @dpc any way to get this to work with svc?

It's been years since I used it, but there's nothing here that would work with about anything. Instead of starting github action runner direclty, you need to start this script, and that should work with anything: crontab, systemd, svc.
